File: EmgVisualizer/mobile/src/main/java/it/ncorti/emgvisualizer/myo/MyoCommandList.java
package it.ncorti.emgvisualizer.myo;

/**
 * This class is List of Myo Commands, allowing to
 * [https://github.com/thalmiclabs/myo-bluetooth].
 *
 * @author Naoki
 */
public class MyoCommandList {

    private byte[] send_bytes_data;

    public byte[] sendUnsetData() {
        byte command_data = (byte) 0x01;
        byte payload_data = (byte) 3;
        byte emg_mode = (byte) 0x00;
        byte imu_mode = (byte) 0x00;
        byte class_mode = (byte) 0x00;
        send_bytes_data =
                new byte[]{command_data, payload_data, emg_mode, imu_mode, class_mode};

        return send_bytes_data;
    }

    public byte[] sendVibration3() {
        byte command_vibrate = (byte) 0x03;
        byte payload_vibrate = (byte) 1;
        byte vibrate_type = (byte) 0x03;
        send_bytes_data = new byte[]{command_vibrate, payload_vibrate, vibrate_type};

        return send_bytes_data;
    }

    public byte[] sendEmgOnly() {
        byte command_data = (byte) 0x01;
        byte payload_data = (byte) 3;
        byte emg_mode = (byte) 0x02;
        byte imu_mode = (byte) 0x00;
        byte class_mode = (byte) 0x00;
        send_bytes_data =
                new byte[]{command_data, payload_data, emg_mode, imu_mode, class_mode};

        return send_bytes_data;
    }

    public byte[] sendUnSleep() {
        byte command_sleep_mode = (byte) 0x09;
        byte payload_unlock = (byte) 1;
        byte never_sleep = (byte) 1;
        send_bytes_data = new byte[]{command_sleep_mode, payload_unlock, never_sleep};

        return send_bytes_data;
    }

    public byte[] sendNormalSleep() {
        byte command_sleep_mode = (byte) 0x09;
        byte payload_unlock = (byte) 1;
        byte normal_sleep = (byte) 0;
        send_bytes_data = new byte[]{command_sleep_mode, payload_unlock, normal_sleep};

        return send_bytes_data;
    }
}

File: myonnaise/src/main/java/com/ncorti/myonnaise/MyoCommandList.kt
package com.ncorti.myonnaise

@Suppress("LocalVariableName", "PrivatePropertyName")
/**
 * This class is List of Myo Commands, allowing to
 * [https://github.com/thalmiclabs/myo-bluetooth].
 */
class MyoCommandList {

    private var send_bytes_data: ByteArray? = null

    fun sendUnsetData(): ByteArray? {
        val command_data = 0x01.toByte()
        val payload_data = 3.toByte()
        val emg_mode = 0x00.toByte()
        val imu_mode = 0x00.toByte()
        val class_mode = 0x00.toByte()
        send_bytes_data = byteArrayOf(command_data, payload_data, emg_mode, imu_mode, class_mode)

        return send_bytes_data
    }

    fun sendVibration3(): ByteArray? {
        val command_vibrate = 0x03.toByte()
        val payload_vibrate = 1.toByte()
        val vibrate_type = 0x03.toByte()
        send_bytes_data = byteArrayOf(command_vibrate, payload_vibrate, vibrate_type)

        return send_bytes_data
    }

    fun sendEmgOnly(): ByteArray? {
        val command_data = 0x01.toByte()
        val payload_data = 3.toByte()
        val emg_mode = 0x02.toByte()
        val imu_mode = 0x00.toByte()
        val class_mode = 0x00.toByte()
        send_bytes_data = byteArrayOf(command_data, payload_data, emg_mode, imu_mode, class_mode)

        return send_bytes_data
    }

    fun sendUnSleep(): ByteArray? {
        val command_sleep_mode = 0x09.toByte()
        val payload_unlock = 1.toByte()
        val never_sleep = 1.toByte()
        send_bytes_data = byteArrayOf(command_sleep_mode, payload_unlock, never_sleep)

        return send_bytes_data
    }

    fun sendNormalSleep(): ByteArray? {
        val command_sleep_mode = 0x09.toByte()
        val payload_unlock = 1.toByte()
        val normal_sleep = 0.toByte()
        send_bytes_data = byteArrayOf(command_sleep_mode, payload_unlock, normal_sleep)

        return send_bytes_data
    }
}
